modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/CutMarksToSmilWorkflowOperationHandler.java

```diff
@@ -0,0 +1,319 @@
+   */
+  public void setWorkspace(Workspace workspace) {
+    this.workspace = workspace;
+  }
+
+  /**
+   * The SMIL service to modify SMIL files.
+   */
+  private SmilService smilService;
+  public void setSmilService(SmilService smilService) {
+    this.smilService = smilService;
+  }
+
+  /** JSON Parser */
+  private static final Gson gson = new Gson();
+
+  /** Stores information read from JSON */
+  class Times {
+    private Long begin;
+    private Long duration;
+
+    public Long getStartTime() {
+      return begin;
+    }
+    public void setStartTime(Long startTime) {
+      this.begin = startTime;
+    }
+    public Long getDuration() {
+      return duration;
+    }
+    public void setDuration(Long duration) {
+      this.duration = duration;
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.workflow.api.WorkflowOperationHandler#start(org.opencastproject.workflow.api.WorkflowInstance,
+   *      JobContext)
+   */
+  @Override
+  public WorkflowOperationResult start(final WorkflowInstance workflowInstance, JobContext context)
+          throws WorkflowOperationException {
+    logger.info("Running cut marks to smil workflow operation on workflow {}", workflowInstance.getId());
+
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+    final MediaPackage mediaPackage = (MediaPackage) workflowInstance.getMediaPackage().clone();
+
+    // Read config options
+    final MediaPackageElementFlavor jsonFlavor = MediaPackageElementFlavor.parseFlavor(
+            getConfig(operation, SOURCE_JSON_FLAVOR));
+    final MediaPackageElementFlavor targetSmilFlavor = MediaPackageElementFlavor.parseFlavor(
+            getConfig(operation, TARGET_SMIL_FLAVOR));
+
+    String flavorNames = operation.getConfiguration(SOURCE_MEDIA_FLAVORS);
+    final List<MediaPackageElementFlavor> flavors = new ArrayList<MediaPackageElementFlavor>();
+    for (String flavorName : asList(flavorNames))
+      flavors.add(MediaPackageElementFlavor.parseFlavor(flavorName));
+
+    // Target tags
+    String targetTagsOption = StringUtils.trimToNull(operation.getConfiguration(TARGET_TAGS));
+    List<String> targetTags = asList(targetTagsOption);
+
+    // Is there a catalog?
+    Catalog[] catalogs = mediaPackage.getCatalogs(jsonFlavor);
+    if (catalogs.length != 1) {
+      logger.warn("Number of catalogs in the source flavor does not equal one. Skipping...");
+      final WorkflowOperationResult result = createResult(mediaPackage, WorkflowOperationResult.Action.SKIP);
+      return result;
+    }
+
+    // Parse JSON
+    Times[] cutmarks;
+    Catalog jsonWithTimes = catalogs[0];
+    try (BufferedReader bufferedReader = new BufferedReader(new FileReader(getMediaPackageElementPath(jsonWithTimes)))) {
+      cutmarks = gson.fromJson(bufferedReader, Times[].class);
+    } catch (Exception e) {
+      throw new WorkflowOperationException("Could not read JSON: " + e);
```

Comment:
Use the `cause` argument of exception constructors:
```suggestion
      throw new WorkflowOperationException("Could not read JSON", e);
```